src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -731,6 +734,39 @@ final InetSocketAddress open(String host, int port, int timeoutMillis, boolean u
         return (InetSocketAddress) channelSocket.getRemoteSocketAddress();
     }
 
+    @SuppressWarnings("unchecked")
+    private void setSocketOptions(Socket tcpSocket, TDSChannel channel) throws IOException {
+        Method setOptionMethod = null;
+        SocketOption<Integer> keepIdleOption = null;
+        SocketOption<Integer> keepIntervalOption = null;
+
+        try {
+            setOptionMethod = Socket.class.getMethod("setOption", SocketOption.class, Object.class);
+            Class<?> clazz = Class.forName("jdk.net.ExtendedSocketOptions");
+            keepIdleOption = (SocketOption<Integer>) clazz.getDeclaredField("TCP_KEEPIDLE").get(null);
+            keepIntervalOption = (SocketOption<Integer>) clazz.getDeclaredField("TCP_KEEPINTERVAL").get(null);
```

Comment:
These steps should be static that we do once so that we don't have to do the reflection lookup every time we make a socket connection.